cogs/utility.py

```diff
@@ -1293,15 +1337,28 @@ async def permissions_override(self, ctx, command_name: str.lower, *, level_name
             self.bot.config["override_command_level"][command.qualified_name] = level.name
 
             await self.bot.config.update()
-            embed = discord.Embed(
-                title="Success",
-                color=self.bot.main_color,
-                description="Successfully set command permission level for "
-                f"`{command.qualified_name}` to `{level.name}`.",
-            )
+            final = command.qualified_name
+
+        if type_ == "group":
+            for command in set(group.walk_commands()):
+                logger.info(
+                    "Updated command permission level for `%s` to `%s`.",
+                    command.qualified_name,
+                    level.name,
+                )
+                self.bot.config["override_command_level"][command.qualified_name] = level.name
+                await self.bot.config.update()
```

Comment:
```suggestion
            update = False
            for command in set(group.walk_commands()):
                logger.info(
                    "Updated command permission level for `%s` to `%s`.",
                    command.qualified_name,
                    level.name,
                )
                self.bot.config["override_command_level"][command.qualified_name] = level.name
                update = True

            if update:
                await self.bot.config.update()
```

Calling `.config.update()` in the `for ...` loop repetitively could cause hitting the rate limit (at some point).
In my opinion, it would be a good practice if we only call it once after everything is set.

Reply:
Whats the point of 
`update = False`
and then setting it to True every loop and then checking if its True and then updating? 

Wont it just work if the `await self.bot.config.update()` is removed from the loop?

Reply:
`False` before the loop.
Set to `True` only if it enters the loop and after `config` is changed.
There probably will be the case where the group (cog) is empty (without commands) tho, so the `update` will remain `False` and `.bot.config.update()` will not be executed.

Reply:
Ah makes sense!